Frends.Community.RabbitMQ/RabbitMQTask.cs

```diff
@@ -12,88 +12,196 @@ namespace Frends.Community.RabbitMQ
-
-            if (inputParams.Create)
+            IConnection connection = CreateConnection(inputParams.HostName, inputParams.ConnectWithURI);
+            IModel channel = connection.CreateModel();
+            try
             {
-                _channel.QueueDeclare(queue: inputParams.QueueName,
-                                durable: inputParams.Durable,
-                                exclusive: false,
-                                autoDelete: false,
-                                arguments: null);
-
-            }
+                if (inputParams.Create)
+                {
+                    channel.QueueDeclare(queue: inputParams.QueueName,
+                        durable: inputParams.Durable,
+                        exclusive: false,
+                        autoDelete: false,
+                        arguments: null);
+                    channel.ConfirmSelect();
+                }
 
-            IBasicProperties basicProperties = null;
-            if (inputParams.Durable == true)
-            {
+                IBasicProperties basicProperties = null;
 
-                basicProperties = _channel.CreateBasicProperties();
-                basicProperties.Persistent = true;
+                if (inputParams.Durable == true)
+                {
+                    basicProperties = channel.CreateBasicProperties();
+                    basicProperties.Persistent = true;
+                }
+                
+                if (inputParams.WriteMessageCount <= 1)
+                {
+                    channel.BasicPublish(exchange:
+                        inputParams.ExchangeName,
+                        routingKey: inputParams.RoutingKey,
+                        basicProperties: basicProperties,
+                        body: inputParams.Data);
+                    return true;
+                }
 
+                return false;
+            }
+            finally
+            {
+               CloseConnection(channel, connection); 
             }
+        }
+        
+        /// <summary>
+        /// Writes messages into a queue with batch publish. All messages are under transaction. If one the message failes all messages will be rolled back.  
+        /// </summary>
+        /// <param name="inputParams"></param>
+        public static bool WriteBatchMessage([PropertyTab] WriteInputParams inputParams)
+        {
+            IConnection connection = CreateConnection(inputParams.HostName, inputParams.ConnectWithURI);
+            IModel channel = connection.CreateModel();
+            try
+            {
+                if (inputParams.Create)
+                {
+                    channel.QueueDeclare(queue: inputParams.QueueName,
+                        durable: inputParams.Durable,
+                        exclusive: false,
+                        autoDelete: false,
+                        arguments: null);
+                    channel.ConfirmSelect();
```

Comment:
Does this work with TxSelect? The documentation says:
"A transactional channel cannot be put into confirm mode and once a channel is in confirm mode, it cannot be made transactional"

Reply:
Simple write messaging uses now channel.ConfirmSelect() and batch write channel.TxSelect()